clang/test/CIR/CodeGen/asm.c

```diff
@@ -1,32 +1,32 @@
 // RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -fclangir-enable -emit-cir %s -o %t.cir
 // RUN: FileCheck --input-file=%t.cir %s
 
-//CHECK: cir.asm(x86_att, {"" ""})
+//CHECK: cir.asm(x86_att, {"" ""})  : () -> ()
 void empty1() {
   __asm__ volatile("" : : : );
 }
 
-//CHECK: cir.asm(x86_att, {"xyz" ""})
+//CHECK: cir.asm(x86_att, {"xyz" ""})  : () -> () 
 void empty2() {
   __asm__ volatile("xyz" : : : );
 }
 
-//CHECK: cir.asm(x86_att, {"" "=*m,m"})
+//CHECK: cir.asm(x86_att, {"" "=*m,*m"}) %0, %0 : (!cir.ptr<!s32i>, !cir.ptr<!s32i>) -> ()
 void t1(int x) {
   __asm__ volatile("" : "+m"(x));
 }
 
-//CHECK: cir.asm(x86_att, {"" "m"})
+//CHECK: cir.asm(x86_att, {"" "*m"}) %0 : (!cir.ptr<!s32i>) -> ()
 void t2(int x) {
   __asm__ volatile("" : : "m"(x));
 }
 
-//CHECK: cir.asm(x86_att, {"" "=*m"})
+//CHECK: cir.asm(x86_att, {"" "=*m"}) %0 : (!cir.ptr<!s32i>) -> ()
 void t3(int x) {
   __asm__ volatile("" : "=m"(x));
 }
 
-//CHECK: cir.asm(x86_att, {"" "=&r,=&r,1"})
+//CHECK: cir.asm(x86_att, {"" "=&r,=&r,1"}) %1 : (!s32i) -> ()
```

Comment:
It's a bit confusing that operands go after the closing `)`. This would probably be a bit more readable:

```
cir.asm(x86_att, {"" "=&r,=&r,1"}, %1, %2 : (!s32i, ...)) -> ()
or
cir.asm(x86_att, {"" "=&r,=&r,1"}, %1 : !s32i, %2 : ...) -> ()
or
cir.asm(x86_att, {"" "=&r,=&r,1"}, %1) : (!s32i) -> ()
or
cir.asm(x86_att, {"" "=&r,=&r,1"}, inputs (%1) : (!s32i), ...) -> () // this could perhaps use oilist
```

Reply:
I guess this depends a bit on what you add next, so probably not worth spending to much time now.

Reply:
I'm gonna go ahead and approve/merge, with the expectation that before we wrap this feature, we have a nice looking `cir.asm`, please flag to me when you are in the last PR of the series, so perhaps we can bikeshed more at the final stage!